relay-general/src/protocol/measurements.rs

```diff
@@ -0,0 +1,127 @@
+pub struct Measurements(pub Object<Measurement>);
+
+impl Measurements {
+    pub fn merge(&mut self, input: Annotated<Measurements>) {
+        let input: Measurements = match input.into_value() {
+            None => {
+                return;
+            }
+            Some(measurements) => measurements,
+        };
+
+        if self.0.is_empty() {
+            self.0 = input.0;
+            return;
+        }
+
+        for (name, measurement) in input.0.into_iter() {
+            self.0.insert(name, measurement);
+        }
+    }
+}
+
+impl FromValue for Measurements {
+    fn from_value(value: Annotated<Value>) -> Annotated<Self> {
+        match value {
+            Annotated(Some(Value::Object(items)), mut meta) => {
+                let mut measurements = Object::<Measurement>::new();
+
+                for (raw_name, raw_value) in items.into_iter() {
+                    let value: Annotated<f64> = match raw_value {
+                        Annotated(Some(Value::Object(bag)), mut object_meta) => {
+                            match bag.get("value") {
+                                Some(Annotated(Some(Value::I64(value)), meta)) => {
+                                    Annotated(Some(*value as f64), meta.clone())
+                                }
+                                Some(Annotated(Some(Value::U64(value)), meta)) => {
+                                    Annotated(Some(*value as f64), meta.clone())
+                                }
+                                Some(Annotated(Some(Value::F64(value)), meta)) => {
+                                    Annotated(Some(*value), meta.clone())
+                                }
+                                Some(Annotated(value, meta)) => {
+                                    let mut meta = meta.clone();
+                                    meta.add_error(Error::expected("number"));
+                                    meta.set_original_value(value.clone());
+                                    Annotated(None, meta)
+                                }
+                                None => {
+                                    object_meta.add_error(Error::expected("value"));
+                                    Annotated(None, object_meta)
+                                }
+                            }
+                        }
+                        Annotated(value, meta) => {
+                            let mut meta = meta.clone();
+                            meta.add_error(Error::expected("object"));
+                            meta.set_original_value(value);
+                            Annotated(None, meta.clone())
+                        }
+                    };
+
+                    let measurement_name = raw_name.to_lowercase();
+
+                    if !MEASUREMENT_NAME.is_match(&measurement_name) {
+                        meta.add_error(Error::expected(
+                            "measurement name to contain only characters a-zA-Z0-9-_.",
```

Comment:
This is out-of-sync with the definition of `MEASUREMENT_NAME`: there, we're not accounting for uppercase ASCII.

Reply:
@rhcarvalho good catch!

Reply:
Should be fixed in https://github.com/getsentry/relay/pull/724/commits/0af9da3eaff23523b8d1a283b720d6b490c5e0ba